shared/crypto/input/index.tsx

```diff
@@ -176,17 +189,117 @@ export const FileInput = (props: FileProps) => {
   )
 }
 
-export const OperationBanner = React.memo((props: OperationBannerProps) => {
-  const color = props.type === 'error' ? 'red' : props.type === 'warning' ? 'yellow' : 'grey'
+export const Input = (props: InputProps) => {
+  const {fileDroppedCounter, operation} = props
+  const dispatch = Container.useDispatch()
+
+  // Store
+  const input = Container.useSelector(state => state.crypto[operation].input.stringValue())
+  const inputType = Container.useSelector(state => state.crypto[operation].inputType)
+
+  // State
+  const [inputValue, setInputValue] = React.useState(input)
+
+  // Actions
+  const onSetInput = React.useCallback(
+    (type: Types.InputTypes, newValue: string) => {
+      dispatch(CryptoGen.createSetInput({operation, type, value: new HiddenString(newValue)}))
+    },
+    [dispatch, operation]
+  )
+  const onClearInput = React.useCallback(() => {
+    dispatch(CryptoGen.createClearInput({operation}))
+  }, [dispatch, operation])
+
+  // Clear the local input state when a user has dragged and dropped a file into the operation
+  // If the input is not cleared then dropping a file, then clearing the file will show old text input
+  const prevCounter = Container.usePrevious(fileDroppedCounter)
```

Comment:
do you even need to keep track like this? if you set the dep to fileDroppedCounter and only clear if its >0 (aka ignore the mount or whatever), then doesn't it work?